sky/clouds/service_catalog/gcp_catalog.py

```diff
@@ -433,6 +448,30 @@ def list_accelerators(
                     price=info.price + vm_price,
                     spot_price=info.spot_price + vm_spot_price,
                 ))
+    # Sort the instances again after adding the vm price, as the order may
+    # have changed.
+    for acc_name, acc_infos in new_infos.items():
+        acc_infos.sort(
+            key=lambda info: (info.accelerator_count, info.instance_type, info.
+                              cpu_count if not pd.isna(info.cpu_count) else 0,
+                              info.price, info.spot_price, info.region))
+        if not all_regions:
+            # Only keep the cheapest instance type across all regions.
+            new_acc_infos = []
+            cur_key = None
+
+            def _get_key(info):
+                return (info.accelerator_name, info.accelerator_count,
+                        info.instance_type,
+                        info.cpu_count if not pd.isna(info.cpu_count) else 0)
+
+            for info in acc_infos:
+                info_key = _get_key(info)
+                if cur_key != info_key:
+                    cur_key = info_key
+                    new_acc_infos.append(info)
```

Comment:
Seems like we are keeping the first entry in the list. Why is that the cheapest?

Reply:
Because we have sorted it by the price in L454. : )